buildcontext/git.go

```diff
@@ -310,6 +342,12 @@ func (gr *gitResolver) resolveGitProject(ctx context.Context, gwClient gwclient.
 		if err != nil {
 			return nil, errors.Wrap(err, "read git-refs")
 		}
+		earthfilePathsRaw, err := gitMetaRef.ReadFile(ctx, gwclient.ReadRequest{
```

Comment:
Just realizing that at some point it might be better to put all this metadata in a single file, to minimize the back-and-forth between earthly and buildkit with all these `ReadFile` requests. In any case, that can be a change that we make in a different PR.